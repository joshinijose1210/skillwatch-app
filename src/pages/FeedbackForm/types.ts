export type DataType = {
    employeeId: string;
    firstName: string;
    lastName: string;
};

export type GetDescriptionType = {
    feedbackTag: string;
};

export interface DescriptionProps {
    heading: string;
    subHeading: string;
    isFeedbackTips: boolean;
    tips?: string[];
    feedbackTips?: { whatToDo: string[]; whatNotToDo: string[] };
    exampleText?: string;
    isItallic?: boolean;
    slackShortcut?: string;
}
