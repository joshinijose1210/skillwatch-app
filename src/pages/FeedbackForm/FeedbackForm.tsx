import { ScreenWidthWrapper, StyledTipsTitle } from '@common';
import { PageContent, RichTextEditor, SelectEmployee, SelectFeedbackTag } from '@components';
import ListHeader from '@components/reusableComponents/ListHeader';
import { Button } from '@medly-components/core';
import { SendIcon } from '@medly-components/icons';
import 'quill/dist/quill.snow.css';
import * as Styled from './FeedbackForm.styled';
import GetDescription from './GetDescription';
import { useFeedbackForm } from './useFeedbackForm';

export const FeedbackForm: React.FC = () => {
    const {
        state,
        selectedEmployee,
        isLoading,
        handleSelectEmployee,
        feedbackTag,
        handleDropdownChange,
        feedbackData,
        handleSend,
        feedbackTagText,
        buttonLoader,
        handleChange,
        isDraftDisabled
    } = useFeedbackForm();

    return (
        <PageContent>
            <ListHeader title={state?.action ? `${state.action} Feedback` : 'Add Feedback'} />
            <Styled.StyledContainer>
                <ScreenWidthWrapper className="form-division">
                    <SelectEmployee disabled={state && state.action == 'View'} value={selectedEmployee} onChange={handleSelectEmployee} />
                    <SelectFeedbackTag
                        disabled={state && state.action == 'View'}
                        feedbackTagValue={feedbackTag}
                        setFeedbackTagValue={handleDropdownChange}
                    />
                    <RichTextEditor
                        disabled={state && state.action === 'View'}
                        editor={'feedback'}
                        input={feedbackData}
                        setInput={handleChange}
                        feedbackTag={feedbackTag}
                    />
                    <Styled.FlexBoxColumn>
                        {((state && state.action === 'Add') || state.action === 'Edit') && (
                            <Styled.SendButtonBox>
                                <Button
                                    data-testid="saveAsDraftButton"
                                    isLoading={buttonLoader === 1 && isLoading}
                                    variant="outlined"
                                    onClick={() => handleSend('Draft')}
                                    disabled={isDraftDisabled}
                                >
                                    Save as Draft
                                </Button>
                                <Button
                                    variant="solid"
                                    data-testid="sendButton"
                                    isLoading={buttonLoader === 2 && isLoading}
                                    onClick={() => handleSend('Save')}
                                    disabled={isDraftDisabled}
                                >
                                    <SendIcon size="S" />
                                    Send
                                </Button>
                            </Styled.SendButtonBox>
                        )}
                    </Styled.FlexBoxColumn>
                </ScreenWidthWrapper>
                <ScreenWidthWrapper className="tips-division">
                    {Boolean(feedbackTagText) && state && state.action !== 'View' && (
                        <Styled.TipContainer feedbackTag={feedbackTag}>
                            <Styled.TipHeaderText>
                                <StyledTipsTitle>Tips</StyledTipsTitle>
                                <Styled.BulbImg />
                            </Styled.TipHeaderText>
                            <GetDescription feedbackTag={feedbackTagText || ''} />
                        </Styled.TipContainer>
                    )}
                </ScreenWidthWrapper>
            </Styled.StyledContainer>
        </PageContent>
    );
};
