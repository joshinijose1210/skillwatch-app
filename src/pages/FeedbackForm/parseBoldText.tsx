// converts **text** to <strong>text</strong>
export const parseBoldText = (text: string) => {
    if (!text) return '';

    try {
        if (typeof text !== 'string') return text;

        const parts = text.split(/(\*\*[^*]+\*\*)/g);
        return parts.map((part, index) => {
            if (part.startsWith('**') && part.endsWith('**')) {
                return <strong key={index}>{part.slice(2, -2)}</strong>;
            }
            return <span key={index}>{part}</span>;
        });
    } catch {
        return text;
    }
};
