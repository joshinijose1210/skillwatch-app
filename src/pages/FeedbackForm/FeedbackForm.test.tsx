import { RTKRender, cleanup, fireEvent, screen, waitFor } from '@utils/test-utils';
import { FeedbackForm } from './FeedbackForm';
import { routeConstants } from '@constants';
import * as reactRouterDom from 'react-router-dom';
import { myFeedbackFormMockData } from '@mocks/myFeedback';
import MockAdapter from 'axios-mock-adapter';
import { apiInstance } from '@utils';
import { AddFeedbackMockData } from '@mocks/addFeedback';

const mockNavigate = jest.fn();
const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
    useLocation: jest.fn()
}));

const useLocationMock = jest.spyOn(reactRouterDom, 'useLocation');

beforeEach(() => {
    useLocationMock.mockReturnValue({
        ...myFeedbackFormMockData.locationMockData,
        pathname: `${routeConstants.myFeedback}/1/add-feedback`,
        state: { ...myFeedbackFormMockData.mockState }
    });
    mock.reset();
});

afterEach(cleanup);

describe('Feedback component', () => {
    it('should take snapshot', async () => {
        const { container } = RTKRender(<FeedbackForm />);
        expect(container).toMatchSnapshot();
    });

    it('should fill in data and send when save button is clicked', async () => {
        useLocationMock.mockReturnValue({
            ...myFeedbackFormMockData.locationMockData,
            pathname: `${routeConstants.myFeedback}/1/add-feedback`,
            state: { ...myFeedbackFormMockData.mockState }
        });
        mock.onGet(AddFeedbackMockData.feedbackData.get.url).replyOnce(
            AddFeedbackMockData.feedbackData.get.responseCode,
            AddFeedbackMockData.feedbackData.get.data
        );
        mock.onGet(AddFeedbackMockData.employeesListData.get.url).replyOnce(
            AddFeedbackMockData.employeesListData.get.responseCode,
            AddFeedbackMockData.employeesListData.get.data
        );
        mock.onPost(AddFeedbackMockData.post.save.url).replyOnce(200, {});
        RTKRender(<FeedbackForm />);
        expect(screen.getByText('Add Feedback')).toBeInTheDocument();
        const feedbackTypeDropdown = await screen.findByLabelText('Feedback Type');
        expect(screen.getByText('Add Feedback')).toBeInTheDocument();
        const employeeName = await screen.findByPlaceholderText<HTMLInputElement>('Select Employee Name');
        const editor = screen.getByTestId('textquill');
        const saveAsDraftButton = screen.getByTestId('saveAsDraftButton');
        const sendButton = screen.getByTestId('sendButton');
        const text = 'jadsbbhshabjhjdasjadsbbhshabjhjdasjadsbbhshabjhjdasjadsbbhshabjhjdas';
        const pElement = editor.querySelector('.ql-editor > p') as HTMLParagraphElement;
        expect(employeeName).toBeInTheDocument();
        expect(feedbackTypeDropdown).toBeInTheDocument();
        expect(pElement).toBeInTheDocument();
        expect(saveAsDraftButton).toBeInTheDocument();
        expect(saveAsDraftButton).toBeDisabled();
        fireEvent.change(pElement, { target: { textContent: text } });
        await waitFor(() => expect(pElement).toHaveTextContent(text));
        fireEvent.click(employeeName);
        const employeeOptions = await screen.findAllByText('User Test (SR003)');
        fireEvent.click(employeeOptions[0]);
        await waitFor(() => expect(employeeName.value).toBe('User Test (SR003)'));
        await waitFor(() => expect(saveAsDraftButton).toBeEnabled());
        fireEvent.click(sendButton);
        expect(await screen.findByText('Feedback submitted successfully')).toBeInTheDocument();
    });

    it('should show error toast when fill in data and send when save button is clicked but fails', async () => {
        useLocationMock.mockReturnValue({
            ...myFeedbackFormMockData.locationMockData,
            pathname: `${routeConstants.myFeedback}/1/add-feedback`,
            state: { ...myFeedbackFormMockData.mockState }
        });
        mock.onGet(AddFeedbackMockData.feedbackData.get.url).replyOnce(
            AddFeedbackMockData.feedbackData.get.responseCode,
            AddFeedbackMockData.feedbackData.get.data
        );
        mock.onGet(AddFeedbackMockData.employeesListData.get.url).replyOnce(
            AddFeedbackMockData.employeesListData.get.responseCode,
            AddFeedbackMockData.employeesListData.get.data
        );
        mock.onPost(AddFeedbackMockData.post.save.url).replyOnce(404, { message: 'something went wrong' });
        RTKRender(<FeedbackForm />);
        const feedbackTypeDropdown = await screen.findByLabelText('Feedback Type');
        const employeeName = await screen.findByPlaceholderText<HTMLInputElement>('Select Employee Name');
        const editor = screen.getByTestId('textquill');
        const saveAsDraftButton = screen.getByTestId('saveAsDraftButton');
        const sendButton = screen.getByTestId('sendButton');
        const text = 'jadsbbhshabjhjdasjadsbbhshabjhjdasjadsbbhshabjhjdasjadsbbhshabjhjdas';
        const pElement = editor.querySelector('.ql-editor > p') as HTMLParagraphElement;
        expect(employeeName).toBeInTheDocument();
        expect(feedbackTypeDropdown).toBeInTheDocument();
        expect(pElement).toBeInTheDocument();
        expect(saveAsDraftButton).toBeInTheDocument();
        expect(saveAsDraftButton).toBeDisabled();
        fireEvent.change(pElement, { target: { textContent: text } });
        await waitFor(() => expect(pElement).toHaveTextContent(text));
        fireEvent.click(employeeName);
        const employeeOptions = await screen.findAllByText('User Test (SR003)');
        fireEvent.click(employeeOptions[0]);
        await waitFor(() => expect(employeeName.value).toBe('User Test (SR003)'));
        await waitFor(() => expect(saveAsDraftButton).toBeEnabled());
        fireEvent.click(sendButton);
        expect(await screen.findByText('something went wrong')).toBeInTheDocument();
    });

    it('should show success toast when fill in data and save as draft when saveAsDraft button is clicked is succesful', async () => {
        useLocationMock.mockReturnValue({
            ...myFeedbackFormMockData.locationMockData,
            pathname: `${routeConstants.myFeedback}/1/add-feedback`,
            state: { ...myFeedbackFormMockData.mockState, action: 'Edit', srNo: 1 }
        });
        mock.onGet(AddFeedbackMockData.feedbackData.get.url).replyOnce(
            AddFeedbackMockData.feedbackData.get.responseCode,
            AddFeedbackMockData.feedbackData.get.data
        );
        mock.onGet(AddFeedbackMockData.employeesListData.get.url).replyOnce(
            AddFeedbackMockData.employeesListData.get.responseCode,
            AddFeedbackMockData.employeesListData.get.data
        );
        mock.onPut(AddFeedbackMockData.post.saveAsDraft.url).replyOnce(200, {});
        RTKRender(<FeedbackForm />);
        expect(await screen.findByText('Edit Feedback')).toBeInTheDocument();
        const feedbackTypeDropdown = await screen.findByLabelText('Feedback Type');
        const employeeName = await screen.findByPlaceholderText<HTMLInputElement>('Select Employee Name');
        const editor = screen.getByTestId('textquill');
        const saveAsDraftButton = screen.getByTestId('saveAsDraftButton');
        const text = 'this is the test feedback that should have atleast 50 characters.';
        const pElement = editor.querySelector('.ql-editor > p') as HTMLParagraphElement;
        expect(employeeName).toBeInTheDocument();
        expect(feedbackTypeDropdown).toBeInTheDocument();
        expect(pElement).toBeInTheDocument();
        expect(saveAsDraftButton).toBeInTheDocument();
        expect(saveAsDraftButton).toBeDisabled();
        fireEvent.change(pElement, { target: { textContent: text } });
        await waitFor(() => expect(pElement).toHaveTextContent(text));
        fireEvent.click(employeeName);
        fireEvent.change(feedbackTypeDropdown, { target: { value: 'Appreciation' } });
        const employeeOptions = await screen.findAllByText('User Test (SR003)');
        fireEvent.click(employeeOptions[0]);
        await waitFor(() => expect(employeeName.value).toBe('User Test (SR003)'));
        await waitFor(() => expect(saveAsDraftButton).toBeEnabled());
        fireEvent.click(saveAsDraftButton);
        expect(await screen.findByText('Feedback saved successfully')).toBeInTheDocument();
    });

    it('should show error toast when fill in data and save as draft when saveAsDraft button is clicked but api fails to save', async () => {
        useLocationMock.mockReturnValue({
            ...myFeedbackFormMockData.locationMockData,
            pathname: `${routeConstants.myFeedback}/1/add-feedback`,
            state: { ...myFeedbackFormMockData.mockState, action: 'Edit', srNo: 1 }
        });
        mock.onGet(AddFeedbackMockData.feedbackData.get.url).replyOnce(
            AddFeedbackMockData.feedbackData.get.responseCode,
            AddFeedbackMockData.feedbackData.get.data
        );
        mock.onGet(AddFeedbackMockData.employeesListData.get.url).replyOnce(
            AddFeedbackMockData.employeesListData.get.responseCode,
            AddFeedbackMockData.employeesListData.get.data
        );
        mock.onPut(AddFeedbackMockData.post.saveAsDraft.url).replyOnce(404, { message: 'something didnt work' });
        RTKRender(<FeedbackForm />);
        expect(await screen.findByText('Edit Feedback')).toBeInTheDocument();
        const feedbackTypeDropdown = await screen.findByLabelText('Feedback Type');
        const employeeName = await screen.findByPlaceholderText<HTMLInputElement>('Select Employee Name');
        const editor = screen.getByTestId('textquill');
        const saveAsDraftButton = screen.getByTestId('saveAsDraftButton');
        const text = 'this is the test feedback that should have atleast 50 characters.';
        const pElement = editor.querySelector('.ql-editor > p') as HTMLParagraphElement;
        expect(employeeName).toBeInTheDocument();
        expect(feedbackTypeDropdown).toBeInTheDocument();
        expect(pElement).toBeInTheDocument();
        expect(saveAsDraftButton).toBeInTheDocument();
        expect(saveAsDraftButton).toBeDisabled();
        fireEvent.change(pElement, { target: { textContent: text } });
        await waitFor(() => expect(pElement).toHaveTextContent(text));
        fireEvent.click(employeeName);
        fireEvent.change(feedbackTypeDropdown, { target: { value: 'Appreciation' } });
        const employeeOptions = await screen.findAllByText('User Test (SR003)');
        fireEvent.click(employeeOptions[0]);
        await waitFor(() => expect(employeeName.value).toBe('User Test (SR003)'));
        await waitFor(() => expect(saveAsDraftButton).toBeEnabled());
        fireEvent.click(saveAsDraftButton);
        expect(await screen.findByText('something didnt work')).toBeInTheDocument();
    });

    it('should fetch and fill in data for edit form', async () => {
        useLocationMock.mockReturnValue({
            ...myFeedbackFormMockData.locationMockData,
            pathname: `${routeConstants.myFeedback}/1/add-feedback`,
            state: {
                ...myFeedbackFormMockData.mockState,
                action: 'Edit',
                feedbackTypeId: 2,
                feedbackType: 'Improvement',
                feedback: 'this is feedback test while edit state',
                actionFrom: 'ReceivedFeedback',
                feedbackFromId: 134
            }
        });
        mock.onGet(AddFeedbackMockData.feedbackData.get.url).replyOnce(
            AddFeedbackMockData.feedbackData.get.responseCode,
            AddFeedbackMockData.feedbackData.get.data
        );
        mock.onGet(AddFeedbackMockData.employeesListData.get.url).replyOnce(
            AddFeedbackMockData.employeesListData.get.responseCode,
            AddFeedbackMockData.employeesListData.get.data
        );
        RTKRender(<FeedbackForm />);
        const feedbackTypeDropdown = screen.getByPlaceholderText<HTMLInputElement>('Select Feedback Type');
        const editor = screen.getByTestId('textquill');
        const pElement = editor.querySelector('.ql-editor > p') as HTMLParagraphElement;
        await waitFor(() => expect(pElement).toHaveTextContent(/this is feedback test while edit state/i));
        await waitFor(() => expect(feedbackTypeDropdown).toHaveValue('Improvement'));
    });
});
