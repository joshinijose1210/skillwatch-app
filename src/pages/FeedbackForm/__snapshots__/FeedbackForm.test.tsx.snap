// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback component should take snapshot 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          Add Feedback
        </h3>
      </div>
    </div>
    <div
      class="sc-murh6i-4 jvXWjR"
    >
      <div
        class="sc-tkiih6-9 bFDgcw form-division"
      >
        <div
          class="sc-tpub7j-0 iWOTdT"
        >
          <div
            class="sc-1hcsztp-0 cwjrcY sc-tkiih6-13 cecHDR"
            id="employee name-wrapper"
            minwidth="100%"
            variant="outlined"
          >
            <div
              class="sc-17d4s39-0 khebHA"
              id="employee name-input-wrapper"
              minwidth="100%"
            >
              <div
                class="sc-19px6zp-0 BrSGZ"
                filled="[object Object]"
                fusion="[object Object]"
                minrows="1"
                outlined="[object Object]"
                textvariant="[object Object]"
                variant="outlined"
              >
                <div
                  class="sc-lhr232-0 btNOKw"
                  variant="outlined"
                >
                  <input
                    aria-describedby="employee name-helper-text"
                    autocomplete="off"
                    class="sc-hb99wz-0 jaEbhp"
                    data-tooltip-id="employee name-input"
                    errortext=""
                    id="employee name-input"
                    inputsize="M"
                    placeholder="Select Employee Name"
                    type="text"
                    value=""
                    variant="outlined"
                  />
                  <label
                    class="sc-1du51u0-0 eMZrST"
                    for="employee name-input"
                    inputwidth="0"
                    variant="outlined"
                  >
                    Employee Name
                  </label>
                </div>
                <svg
                  class="sc-1pf3x8a-0 klDMsE sc-dynlvq-0 bdmviR"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                    fill="#000"
                  />
                </svg>
              </div>
              
            </div>
          </div>
          
        </div>
        <div
          class="sc-ayn6ri-0 boWEAe"
        >
          <div
            class="sc-1hcsztp-0 fjPkSb sc-tkiih6-13 cecHDR"
            id="feedback type-wrapper"
            minwidth="100%"
            variant="outlined"
          >
            <div
              class="sc-17d4s39-0 khebHA"
              id="feedback type-input-wrapper"
              minwidth="100%"
            >
              <div
                class="sc-19px6zp-0 BrSGZ"
                filled="[object Object]"
                fusion="[object Object]"
                minrows="1"
                outlined="[object Object]"
                textvariant="[object Object]"
                variant="outlined"
              >
                <div
                  class="sc-lhr232-0 btNOKw"
                  variant="outlined"
                >
                  <input
                    aria-describedby="feedback type-helper-text"
                    autocomplete="off"
                    class="sc-hb99wz-0 jaEbhp"
                    data-tooltip-id="feedback type-input"
                    errortext=""
                    id="feedback type-input"
                    inputmode="none"
                    inputsize="M"
                    placeholder="Select Feedback Type"
                    type="text"
                    value=""
                    variant="outlined"
                  />
                  <label
                    class="sc-1du51u0-0 eMZrST"
                    for="feedback type-input"
                    inputwidth="0"
                    variant="outlined"
                  >
                    Feedback Type
                  </label>
                </div>
                <svg
                  class="sc-1pf3x8a-0 klDMsE sc-dynlvq-0 bdmviR"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                    fill="#000"
                  />
                </svg>
              </div>
              
            </div>
          </div>
          
        </div>
        <div
          class="sc-1wh81lg-1 cwMleO"
        >
          <div
            class="sc-1wh81lg-0 RMuwJ"
            editor="feedback"
          >
            <div
              class="ql-toolbar ql-snow"
            >
              <span
                class="ql-formats"
              >
                <button
                  class="ql-bold"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <path
                      class="ql-stroke"
                      d="M5,4H9.5A2.5,2.5,0,0,1,12,6.5v0A2.5,2.5,0,0,1,9.5,9H5A0,0,0,0,1,5,9V4A0,0,0,0,1,5,4Z"
                    />
                     
                    <path
                      class="ql-stroke"
                      d="M5,9h5.5A2.5,2.5,0,0,1,13,11.5v0A2.5,2.5,0,0,1,10.5,14H5a0,0,0,0,1,0,0V9A0,0,0,0,1,5,9Z"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-italic"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="13"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="5"
                      x2="11"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="8"
                      x2="10"
                      y1="14"
                      y2="4"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-underline"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <path
                      class="ql-stroke"
                      d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
                    />
                     
                    <rect
                      class="ql-fill"
                      height="1"
                      rx="0.5"
                      ry="0.5"
                      width="12"
                      x="3"
                      y="15"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-strike"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke ql-thin"
                      x1="15.5"
                      x2="2.5"
                      y1="8.5"
                      y2="9.5"
                    />
                     
                    <path
                      class="ql-fill"
                      d="M9.007,8C6.542,7.791,6,7.519,6,6.5,6,5.792,7.283,5,9,5c1.571,0,2.765.679,2.969,1.309a1,1,0,0,0,1.9-.617C13.356,4.106,11.354,3,9,3,6.2,3,4,4.538,4,6.5a3.2,3.2,0,0,0,.5,1.843Z"
                    />
                     
                    <path
                      class="ql-fill"
                      d="M8.984,10C11.457,10.208,12,10.479,12,11.5c0,0.708-1.283,1.5-3,1.5-1.571,0-2.765-.679-2.969-1.309a1,1,0,1,0-1.9.617C4.644,13.894,6.646,15,9,15c2.8,0,5-1.538,5-3.5a3.2,3.2,0,0,0-.5-1.843Z"
                    />
                     
                  </svg>
                </button>
              </span>
              <span
                class="ql-formats"
              >
                <span
                  class="ql-align ql-picker ql-icon-picker"
                >
                  <span
                    aria-controls="ql-picker-options-0"
                    aria-expanded="false"
                    class="ql-picker-label"
                    role="button"
                    tabindex="0"
                  >
                    <svg
                      viewBox="0 0 18 18"
                    >
                       
                      <line
                        class="ql-stroke"
                        x1="3"
                        x2="15"
                        y1="9"
                        y2="9"
                      />
                       
                      <line
                        class="ql-stroke"
                        x1="3"
                        x2="13"
                        y1="14"
                        y2="14"
                      />
                       
                      <line
                        class="ql-stroke"
                        x1="3"
                        x2="9"
                        y1="4"
                        y2="4"
                      />
                       
                    </svg>
                  </span>
                  <span
                    aria-hidden="true"
                    class="ql-picker-options"
                    id="ql-picker-options-0"
                    tabindex="-1"
                  >
                    <span
                      class="ql-picker-item ql-selected"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        viewBox="0 0 18 18"
                      >
                         
                        <line
                          class="ql-stroke"
                          x1="3"
                          x2="15"
                          y1="9"
                          y2="9"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="3"
                          x2="13"
                          y1="14"
                          y2="14"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="3"
                          x2="9"
                          y1="4"
                          y2="4"
                        />
                         
                      </svg>
                    </span>
                    <span
                      class="ql-picker-item"
                      data-value="center"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        viewBox="0 0 18 18"
                      >
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="3"
                          y1="9"
                          y2="9"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="14"
                          x2="4"
                          y1="14"
                          y2="14"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="12"
                          x2="6"
                          y1="4"
                          y2="4"
                        />
                         
                      </svg>
                    </span>
                    <span
                      class="ql-picker-item"
                      data-value="right"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        viewBox="0 0 18 18"
                      >
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="3"
                          y1="9"
                          y2="9"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="5"
                          y1="14"
                          y2="14"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="9"
                          y1="4"
                          y2="4"
                        />
                         
                      </svg>
                    </span>
                    <span
                      class="ql-picker-item"
                      data-value="justify"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        viewBox="0 0 18 18"
                      >
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="3"
                          y1="9"
                          y2="9"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="3"
                          y1="14"
                          y2="14"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="3"
                          y1="4"
                          y2="4"
                        />
                         
                      </svg>
                    </span>
                  </span>
                </span>
                <select
                  class="ql-align"
                  style="display: none;"
                >
                  <option
                    selected="selected"
                  />
                  <option
                    value="center"
                  />
                  <option
                    value="right"
                  />
                  <option
                    value="justify"
                  />
                </select>
              </span>
              <span
                class="ql-formats"
              >
                <button
                  class="ql-list"
                  type="button"
                  value="ordered"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="15"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="15"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="15"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke ql-thin"
                      x1="2.5"
                      x2="4.5"
                      y1="5.5"
                      y2="5.5"
                    />
                     
                    <path
                      class="ql-fill"
                      d="M3.5,6A0.5,0.5,0,0,1,3,5.5V3.085l-0.276.138A0.5,0.5,0,0,1,2.053,3c-0.124-.247-0.023-0.324.224-0.447l1-.5A0.5,0.5,0,0,1,4,2.5v3A0.5,0.5,0,0,1,3.5,6Z"
                    />
                     
                    <path
                      class="ql-stroke ql-thin"
                      d="M4.5,10.5h-2c0-.234,1.85-1.076,1.85-2.234A0.959,0.959,0,0,0,2.5,8.156"
                    />
                     
                    <path
                      class="ql-stroke ql-thin"
                      d="M2.5,14.846a0.959,0.959,0,0,0,1.85-.109A0.7,0.7,0,0,0,3.75,14a0.688,0.688,0,0,0,.6-0.736,0.959,0.959,0,0,0-1.85-.109"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-list"
                  type="button"
                  value="bullet"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="6"
                      x2="15"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="6"
                      x2="15"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="6"
                      x2="15"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="3"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="3"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="3"
                      y1="14"
                      y2="14"
                    />
                     
                  </svg>
                </button>
              </span>
              <span
                class="ql-formats"
              >
                <button
                  class="ql-link"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="11"
                      y1="7"
                      y2="11"
                    />
                     
                    <path
                      class="ql-even ql-stroke"
                      d="M8.9,4.577a3.476,3.476,0,0,1,.36,4.679A3.476,3.476,0,0,1,4.577,8.9C3.185,7.5,2.035,6.4,4.217,4.217S7.5,3.185,8.9,4.577Z"
                    />
                     
                    <path
                      class="ql-even ql-stroke"
                      d="M13.423,9.1a3.476,3.476,0,0,0-4.679-.36,3.476,3.476,0,0,0,.36,4.679c1.392,1.392,2.5,2.542,4.679.36S14.815,10.5,13.423,9.1Z"
                    />
                     
                  </svg>
                </button>
              </span>
            </div>
            <div
              class="sc-1wh81lg-2 dPewCh ql-container ql-snow"
              data-testid="textquill"
            >
              <div
                class="ql-editor ql-blank"
                contenteditable="true"
                data-gramm="false"
                data-placeholder="Write your feedback here..."
              >
                <p>
                  <br />
                </p>
              </div>
              <div
                class="ql-clipboard"
                contenteditable="true"
                tabindex="-1"
              />
              <div
                class="ql-tooltip ql-hidden"
              >
                <a
                  class="ql-preview"
                  href="about:blank"
                  rel="noopener noreferrer"
                  target="_blank"
                />
                <input
                  data-formula="e=mc^2"
                  data-link="https://quilljs.com"
                  data-video="Embed URL"
                  type="text"
                />
                <a
                  class="ql-action"
                />
                <a
                  class="ql-remove"
                />
              </div>
            </div>
            <div
              class="sc-1wh81lg-3 dQvbgb"
            >
              0/1000
            </div>
            <span
              class="sc-1ntgl3l-0 fbvXZC"
              textalign="initial"
              textcolor="red"
              textvariant="body2"
            />
          </div>
        </div>
        <div
          class="sc-murh6i-2 iBhHAS"
        >
          <div
            class="sc-murh6i-1 cTZoTE"
          >
            <button
              class="sc-1k2druj-0 dsEKPA"
              data-testid="saveAsDraftButton"
              disabled=""
              edges="square"
              type="button"
              variant="outlined"
            >
              <span
                class="sc-1ntgl3l-0 eOsHHD"
                textalign="initial"
                textcolor="inherit"
                textvariant="button1"
              >
                Save as Draft
              </span>
            </button>
            <button
              class="sc-1k2druj-0 eLgfbc"
              data-testid="sendButton"
              disabled=""
              edges="square"
              type="button"
              variant="solid"
            >
              <svg
                class="sc-1pf3x8a-0 bMQNVa"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.671 20.4l17.45-7.48a1 1 0 0 0 0-1.84L3.671 3.6a.993.993 0 0 0-1.39.91l-.01 4.61c0 .5.37.93.87.99L17.271 12l-14.13 1.88c-.5.07-.87.5-.87 1l.01 4.61c0 .71.73 1.2 1.39.91z"
                  fill="#000"
                  fill-opacity=".54"
                />
              </svg>
              <span
                class="sc-1ntgl3l-0 eOsHHD"
                textalign="initial"
                textcolor="inherit"
                textvariant="button1"
              >
                Send
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-tkiih6-9 bFDgcw tips-division"
      />
    </div>
  </main>
</div>
`;
